react-native/react/tracker/index.js

```diff
@@ -1,89 +1,144 @@
 'use strict'
+/* @flow */
 
+// $FlowIssue base-react
 import React, {Component} from '../base-react'
+// $FlowIssue base-redux
 import {connect} from '../base-redux'
+// $FlowIssue platform dependent files
 import Render from './render'
-import {navigateUp} from '../actions/router'
+
+import * as trackerActions from '../actions/tracker'
+import { bindActionCreators } from 'redux'
+
+import type {UserInfo} from './bio.render.desktop'
+import type {Proof} from './proofs.render.desktop'
+import type {SimpleProofState} from '../constants/tracker'
+
+type TrackerProps = {
+  serverStarted: boolean,
+  proofState: SimpleProofState,
+  username: ?string,
+  shouldFollow: ?boolean,
+  reason: string,
+  userInfo: UserInfo,
+  proofs: Array<Proof>,
+  onCloseFromHeader: () => void,
+  onCloseFromActionBar: () => void,
+  onRefollow: () => void,
+  onUnfollow: () => void,
+  onFollowHelp: () => void,
+  onFollowChecked: () => void,
+  registerIdentifyUi: () => void
+}
 
 class Tracker extends Component {
-  constructor (props: any) {
-    super(props)
+  props: TrackerProps;
 
-    // this is TEMP since we don't have a store yet
-    this.state = {
-      shouldFollowChecked: props.shouldFollow
+  componentWillMount () {
+    if (!this.props.serverStarted) {
+      console.log('starting server')
+      this.props.registerIdentifyUi()
     }
   }
 
   render () {
     // these non-prop values will be removed during integration
-    return <Render {...this.props}
-      shouldFollow={this.state.shouldFollowChecked}
-      followChecked={checked => this.setState({shouldFollowChecked: checked})}
-      />
+    return <Render
+             bioProps={{
```

Comment:
is it worth splitting this up like this? it feels like this component shouldn't know that level of detail, it just wants the thing rendered so it can just give the flat props